test/Twig/Tests/Node/DeprecatedTest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+class Twig_Tests_Node_DeprecatedTest extends Twig_Test_NodeTestCase
+{
+    public function testConstructor()
+    {
+        $expr = new Twig_Node_Expression_Constant('foo', 1);
+        $node = new Twig_Node_Deprecated($expr, 1);
+
+        $this->assertEquals($expr, $node->getNode('expr'));
+    }
+
+    public function getTests()
+    {
+        $expr = new Twig_Node_Expression_Constant('foo', 1);
+        $node = new Twig_Node_Deprecated($expr, 1, 'deprecated');
+        $node->setTemplateName('foo.twig');
+
+        return array(array($node, "// line 1\n@trigger_error(\"foo\".' (\"foo.twig\" at line 1).', E_USER_DEPRECATED);"));
```

Comment:
we should have a test compiling the node with some indentation set, to be able to see issues related to use `raw` vs `write`